source/reference/cutover-process.txt

```diff
@@ -164,54 +142,52 @@ Steps
       .. code-block:: javascript
 
          db.adminCommand(
-           {
-             setUserWriteBlockMode: 1,
-             global: false
-           }
+            {
+               setUserWriteBlockMode: 1,
+               global: false
+            }
          )
 
+      Then, transfer your application workload to the destination cluster.
+
    .. step:: Call the ``progress`` endpoint to determine the status of the ``mongosync`` process.
 
-      When the ``mongosync`` progress response indicates that the 
+      When the ``mongosync`` progress response indicates that the
       ``mongosync`` state is ``COMMITTED``, the cutover process is
-      complete. 
+      complete.
 
-      Request
-      ~~~~~~~
+      .. io-code-block::
+         :copyable: true
 
-      .. literalinclude:: /includes/api/requests/progress.sh
-         :language: shell
+         .. input::
+            :language: bash
 
-      Response
-      ~~~~~~~~
+            curl localhost:27182/api/v1/progress -XGET | jq ".progress.state"
 
-      .. code-block:: json
-         :emphasize-lines: 4
-         :copyable: false
+         .. output::
+            :language: json
+
+            "COMMITTED"
+
+Behavior
+--------
+
+.. _c2c-canWrite-committed:
+
+canWrite and COMMITTED
+~~~~~~~~~~~~~~~~~~~~~~
 
-         {
-            "progress":
-               {
-                  "state":"COMMITTED",
-                  "canCommit":true,
-                  "canWrite":false,
-                  "info":"change event application",
-                  "lagTimeSeconds":0,
-                  "collectionCopy":
-                     {
-                        "estimatedTotalBytes":694,
-                        "estimatedCopiedBytes":694
-                     },
-                  "directionMapping":
-                     {
-                        "Source":"cluster0: localhost:27017",
-                        "Destination":"cluster1: localhost:27018"
-                     }
-               },
-            "success": true
-         }
+``mongosync`` permits writes on the destination cluster at much earlier stage than
+the ``COMMITTED`` state.
 
+In the initial sync, ``mongosync`` replicates unique indexes on the source
+cluster as non-unique indexes on the destination cluster. When writes are possible
+on the destination cluster, ``mongosync`` begins to convert these non-unique indexes
+to unique indexes.
 
+This process may take a long time when syncing large collections. Queries
+on the destination cluster that require a unique index may return unexpected
+results.
```

Comment:
If poss, add some info on what would be unexpected.

Reply:
Leaving this up as well for the external.

Reply:
That's a good question. I don't think anything would be unexpected for unconverted unique indexes because the documents should already be in the destination.

Reply:
^ agreed. I'm realizing now that I should've provided a bit more context on the JIRA ticket about this section.

I think we should note that when monogsync indicates canWrite:true, mongosync would've already converted these non-unique indexes to have the `prepareUnique` property. This means that you cannot insert _new_ documents that violate unique index constraints on the destination.

Converting `prepareUnique` to `unique` means that the server performs an index scan to see if any preexisting documents violate unique constraints.

These (unique index [docs](https://www.mongodb.com/docs/manual/core/index-unique/convert-to-unique/#std-label-index-convert-to-unique)) and collMod [docs](https://www.mongodb.com/docs/manual/reference/command/collMod/#mongodb-collflag-index) have more information. It might be helpful to link them here.

Reply:
@ajayvijayakumar123, So the process we should convey:

* During initial sync, unique indexes become non-unique indexes.
* In that process (or after?), the relevant indexes on the destination cluster are set to prepareUnique.
* `/progress` returns `canWrite: true`.
* `mongosync` converts prepareUnique indexes to unique indexes.

Questions:
* If the index scan and conversion to unique raises an error, where would the user look for it?
* If `canWrite: true` happens after the `prepareUnique` value is set, would this mean that the errors originate in duplicates synced from the source cluster?

Reply:
> If `canWrite: true` happens after the `prepareUnique` value is set, would this mean that the errors originate in duplicates synced from the source cluster?

Yes, if there is an error after `canWrite: true`, the duplicates came from the source cluster, assuming that mongosync does not have a bug. In this case, the destination cluster will function identically to the source cluster.

I would recommend we say something like:

"If the commit fails after `canWrite: true`, this means that the source cluster has documents that violate unique index constraints. The destination cluster will not accept any new duplicate values for the failed index constraints, but cannot fully add the constraint due to the duplicates that it has received from the source. If you wish to resolve these duplicates and fully enable the index constraint, manual intervention is required."

Reply:
Apologies for the late response.

The process would:
* During initial sync, unique indexes become non-unique indexes
* _During commit_, the relevant indexes on the destination cluster are set to prepareUnique.
* /progress returns canWrite: true.
* mongosync converts prepareUnique indexes to unique indexes

Good questions. We currently have an engineering proposal (https://jira.mongodb.org/browse/WRITING-25602) on our radar to get consensus on how we're treating these potential errors between canWrite:true and COMMITTED. We want to only provide context on this once the EP is approved. Until then, I think we should only document the differences between canWrite:true and COMMITTED and not the errors that can arise between them. Does that make sense?

Reply:
Fixed